### PR TITLE
Accountant setup

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

In-housing @tingaloo 's outstanding work in #1267 

This pull request makes the following changes:
* Adds accounting functionality

It relates to the following issue #s: 
* Bumps #858 
* Stems from #1267

### CODE REVIEW

File: app/controllers/accountants_controller.rb [hunk 1]
```diff
@@ -1,10 +1,31 @@
 # Controller pertaining to accountant functions, usually fulfillments.
 class AccountantsController < ApplicationController
-  def index; end
+  before_action :confirm_admin_user
+  before_action :find_patient, only: [:edit]
+
+  def index
+    fulfilled_pts = Patient.where('fulfillment.fulfilled' => true)
```
Comment (colinxfleming) [inline app/controllers/accountants_controller.rb]: note to self to scope this a little tighter so we don't light up the db by accident

File: app/controllers/accountants_controller.rb [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+    @patients = Kaminari.paginate_array(fulfilled_pts)
+                        .page(params[:page]).per(25)
+  end
 
   def search
-    @results = Patient.search params[:search]
+    @results = if params[:search] != ''
```
Comment (colinxfleming) [inline app/controllers/accountants_controller.rb]: blank

File: app/views/accountants/_table_content.html.erb [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
       <tr>
-        <th>Patient ID</th>
-        <th>Name</th>
-        <th>Pledge Sent Date</th>
         <th>Clinic Name</th>
-        <th>Clinic Location</th>
+        <th>PT initials</th>
+        <th>Abortion Care $</th>
+        <th>Weeks at Procedure</th>
+        <%= th_autosortable 'Pledge Sent Date', 'string', local_assigns %>
```
Comment (colinxfleming) [inline app/views/accountants/_table_content.html.erb]: not sure this works yet

File: app/views/accountants/edit.js.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+$('.modal-content').html('')
+$('.modal-content').html("<%= escape_javascript(render partial: 'patients/fulfillment', locals: { patient: @patient, styles: 'partial_fulfillment_modal' })%>")
+
+// TODO pass params from search to patients fulfillment partial up here
```
Comment (colinxfleming) [inline app/views/accountants/edit.js.erb]: figure out what this means

File: app/views/accountants/index.html.erb [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+    <div class="row">
+      <%= render partial: 'accountants/search_form' %>    
+    </div>
+
+    <div class="row">
+      <h1 class='border-bottom title'>
+        Accountant Dashboard
+        <small class='pull-right'>
+          <span id='pagination-entries'>
+            <%= page_entries_info @patients, entry_name: 'patient' %>
```
Comment (colinxfleming) [inline app/views/accountants/index.html.erb]: Confirm that this works right. May need to add an itest for this

File: app/views/accountants/search.js.erb [hunk 6]
```diff
@@ -1,5 +1,12 @@
-$('#search_results_shell').empty();
+$('#accounting-list').empty();
 
 <% if @results.any? %>
-  $("#search_results_shell").append("<%= escape_javascript(render partial: 'accountants/table_content', locals: { title: 'Search Results', table_type: 'search_results', patients: @results } ) %>");
+  $("#accounting-list").append("<%= escape_javascript(render partial: 'accountants/table_content', locals: { patients: @results, autosortable: true } ) %>");
 <% end %>
+
+<% patients = Kaminari.paginate_array(@results).page(params[:page]).per(25) %>
```
Comment (colinxfleming) [inline app/views/accountants/search.js.erb]: Probably should do this controllerside

File: test/controllers/accountants_controller_test.rb [hunk 7]
```diff
@@ -24,4 +24,14 @@ class AccountantsControllerTest < ActionDispatch::IntegrationTest
       end
     end
   end
+
```
Comment (colinxfleming) [inline test/controllers/accountants_controller_test.rb]: fill out tests in here more deeply

File: test/integration/accountant_table_search_test.rb [hunk 8]
```diff
@@ -0,0 +1,95 @@
+require 'test_helper'
+
+class AccountantTableSearchTest < ActionDispatch::IntegrationTest
```
Comment (colinxfleming) [inline test/integration/accountant_table_search_test.rb]: oh lord, this predates systemtest. will need to remix this too

File: app/controllers/accountants_controller.rb (hunk 1 above)
Comment (colinxfleming) [reply app/controllers/accountants_controller.rb]: done

File: app/controllers/accountants_controller.rb (hunk 2 above)
Comment (colinxfleming) [reply app/controllers/accountants_controller.rb]: done

File: test/controllers/accountants_controller_test.rb (hunk 7 above)
Comment (colinxfleming) [reply test/controllers/accountants_controller_test.rb]: done

File: app/views/accountants/search.js.erb (hunk 6 above)
Comment (colinxfleming) [reply app/views/accountants/search.js.erb]: done

Comment (colinxfleming) [reply app/views/accountants/search.js.erb]: done

File: app/views/accountants/_table_content.html.erb (hunk 3 above)
Comment (colinxfleming) [reply app/views/accountants/_table_content.html.erb]: done

**Comment by colinxfleming** [comment]
some additional evil magic that's making the pagination div not disappear on empty search results

**Comment by colinxfleming** [comment]
We're ready to go! Tagging in @tingaloo and/or @lomky for review

File: Gemfile [hunk 9]
```diff
@@ -38,6 +38,9 @@ gem 'bootstrap_form'
 # Security libraries
 gem 'rack-attack', '~> 5.1'
 gem 'secure_headers', '~> 3.6', '>= 3.6.4'
+gem 'kaminari-mongoid'
+gem 'kaminari'
```
Comment (lomky) [inline Gemfile]: Minor nitpicks: these aren't security libraries. It'd also be nice to have their versions set.

File: app/models/patient.rb [hunk 10]
```diff
@@ -24,6 +24,8 @@ class Patient
     scope line.downcase.to_sym, -> { where(:line.in => [line]) }
   end
 
+  # paginates_per 1
```
Comment (lomky) [inline app/models/patient.rb]: ?

**Comment by colinxfleming** [comment]
@lomky the UX issue was a great catch. I hooked in some javascript that updates the underlying table, and moves the row to its own partial, so we can overwrite individual rows as needed. Should be good to go now, after tests pass!

**Comment by colinxfleming** [comment]
That's rude, all the system tests are failing for no good reason! Ill square that away and merge after I get things to pass.

**Comment by colinxfleming** [comment]
YUS

**Pull request merged**